projects/wave-core-new/src/lib/project/project.service.spec.ts

```diff
@@ -0,0 +1,110 @@
+import {Project} from './project.model';
+import {ProjectService} from './project.service';
+import {SpatialReferences} from '../operators/spatial-reference.model';
+import {Time} from '../time/time.model';
+import moment from 'moment';
+import {Session} from '../users/session.model';
+import {User} from '../users/user.model';
+import {NEVER, of} from 'rxjs';
+import {Config, WAVE_DEFAULT_CONFIG} from '../config.service';
+import {CreateProjectResponseDict, STRectangleDict, TimeStepDict, UUID} from '../backend/backend.model';
+import {waitForAsync} from '@angular/core/testing';
+import {NotificationService} from '../notification.service';
+import {MapService} from '../map/map.service';
+import {BackendService} from '../backend/backend.service';
+import {UserService} from '../users/user.service';
+
+let notificationServiceSpy: {get: jasmine.Spy};
+let mapServiceSpy: {get: jasmine.Spy};
+let backendSpy: {createProject: jasmine.Spy; listProjects: jasmine.Spy};
+let userServiceSpy: {getSessionStream: jasmine.Spy; getSessionTokenForRequest: jasmine.Spy};
+
+let projectService: ProjectService;
+
+beforeEach(() => {
+    notificationServiceSpy = jasmine.createSpyObj('NotificationService', ['error']);
+    mapServiceSpy = jasmine.createSpyObj('MapService', ['getViewportSizeStream']);
+    backendSpy = jasmine.createSpyObj('BackendService', ['createProject', 'listProjects']);
+    userServiceSpy = jasmine.createSpyObj('UserService', ['getSessionStream', 'getSessionTokenForRequest']);
+
+    // always return the same session
+    userServiceSpy.getSessionStream.and.returnValue(
+        of<Session>({
+            sessionToken: 'session-token',
+            user: new User({
+                id: 'user-id',
+            }),
+            validUntil: moment.utc('2020-01-01'),
```

Comment:
is it a good idea to use a date in the past? It basically means the session is invalid

Reply:
ae20f9d